add agent traces docs (#2492)
huggingface/hub-docs: docs/hub/data-studio.md

<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/datastudio-row-dark.png"/>
</div>

## Agent traces

> Push your raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset (or a [Storage Bucket](./storage-buckets)), and your dataset page will show a dedicated trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./datasets-agent-traces) for supported agents and viewing traces.

Reviewer: ```suggestion
Push your raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset (or a [Storage Bucket](./storage-buckets)) to automatically benefit from a dedicated trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./datasets-agent-traces) for supported agents and viewing traces.
```
it's a bit confusing otherwise imo re. buckets capabilities

> Drop raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset (or a [Storage Bucket](./storage-buckets)), and Data Studio swaps the default table view for a dedicated trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./datasets-agent-traces) for supported agents and viewing traces.

Reviewer 2: ```suggestion
Push your raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset (or a [Storage Bucket](./storage-buckets)), and your dataset page will show a dedicated trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./datasets-agent-traces) for supported agents and viewing traces.
```

> Push your raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset to browse them in Data Studio, or sync them to a [Storage Bucket](./storage-buckets) and open individual `.jsonl` files there. Both paths use the same trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./datasets-agent-traces) for supported agents and viewing traces.

Reviewer 3: ```suggestion
Push your raw JSONL traces from tools like Claude Code, Codex, or Pi into a dataset to browse them in Data Studio, or sync them to a [Storage Bucket](./storage-buckets) and open individual `.jsonl` files there. Both paths use the same trace viewer for stepping through sessions, turns, tool calls, and model responses. See [Agent Traces](./agent-traces) for supported agents and viewing traces.
```

```yaml
---
configs:
  - config_name: default
    data_files:
      - split: train
        path: "data.csv"
      - split: test

>         path: "holdout.csv"

Reviewer 4: minor nit, but I'd leave these untouched; we use the same style elsewhere

<img width="648" height="551" alt="Image" src="https://github.com/user-attachments/assets/356cda62-3678-4e0e-9981-64cdb5589863" />

Author: yes, reverted.
